ufl/algorithms/analysis.py

```diff
@@ -143,8 +143,10 @@ def extract_arguments_and_coefficients(a):
 
 def extract_elements(form):
     "Build sorted tuple of all elements used in form."
-    args = chain(*extract_arguments_and_coefficients(form))
-    return tuple(f.ufl_element() for f in args)
+    arguments, coefficients = extract_arguments_and_coefficients(form)
+    elements_a = tuple(a.ufl_function_space().ufl_element() for a in arguments)
+    elements_c = tuple(c.ufl_element() for c in coefficients)
+    return elements_a + elements_c
```

Comment:
Why don’t coefficients have a function space?

Reply:
It does. A mistake on my behalf.